pino.js

```diff
@@ -96,7 +97,7 @@ function pino (...args) {
   assertDefaultLevelFound(level, customLevels, useOnlyCustomLevels)
   const levels = mappings(customLevels, useOnlyCustomLevels)
 
-  const instance = {
+  Object.assign(instance, {
```

Comment:
can we avoid doing this? It will slow down init time

Reply:
This is called only once and the difference between `Object.assign` and `{}` will be nanoseconds. So Comparing to other code and node itself there will be less than statistical uncertainty.

Reply:
This is just run once.. this might be relevant only for serverless if at all.

Reply:
Actually this has a side effect, as all the definitions of the hidden class would not be done in the same place.
Can you revert?

(Good spot @davidmarkclements).

Reply:
If instance is needed to be passed to normalize function then it should be defined before calling normalize. And normalize returns `opts` that are used for constructing the instance. So I don't know what's the way of avoiding this.

Reply:
I think this might not be needed at all.

In https://github.com/pinojs/pino/blob/cf6cf489d6f0b3f5ded6e27ece10febef3525598/lib/tools.js#L174 you have the last logger object used. You might just call https://github.com/pinojs/pino/blob/cf6cf489d6f0b3f5ded6e27ece10febef3525598/lib/tools.js#L237 with `pretty.call(this.lastLogger, ...)` instead.

Reply:
@SkeLLLa how about resolving this so we can merge?